test/end-to-end/core.sh

```diff
@@ -196,11 +196,32 @@ os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/custom/cross:namespace-p
 echo "[INFO] Cross namespace pull successful"
 
 # check to make sure an image-pusher can push an image
+os::cmd::expect_success "oc project ${CLUSTER_ADMIN_CONTEXT}"
```

Comment:
if we switch projects here, that affects the namespace the next line is adding a role to... I'm having a hard time following the implications your changes have on the existing tests... a lot of those tests depend on the permissions `pusher` has (or doesn't have) in various namespaces.